mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+
+                - 1. `load from file`: Load image metas from pkl file, which
+                is suggested to use. We provided a script to get image metas:
+                `tools/misc/get_image_metas.py`, which need to run this script
+                 before training/testing. Please refer to
+                `config/openimages/README.md` for more details.
+
+                - 2. `load from pipeline`, which will get image metas during
+                test time. However, this may reduce the inference speed,
+                especially when using distribution.
+
+            load_from_file (bool): Whether get image metas from pkl file.
+            meta_file (str): File path to get image metas.
+            filter_labels (bool): Whether filter unannotated classes.
+                Default: True.
+            load_image_level_labels (bool): Whether load and consider image
+                level labels during evaluating. Default: True.
+        """
+
+        self.cat2label = defaultdict(str)
+        self.index_dict = {}
+        # need get `index_dict` before load annotations
+        class_names = self.get_classes_from_csv(label_description_file)
+        super(OpenImagesDataset, self).__init__(**kwargs)
+        self.CLASSES = class_names
+        self.image_level_ann_file = image_level_ann_file
+        self.load_image_level_labels = load_image_level_labels
+        if get_supercategory is True:
+            assert hierarchy_file is not None
+            self.class_label_tree = self.get_relation_matrix(hierarchy_file)
+        self.get_supercategory = get_supercategory
+        self.get_metas = get_metas
+        self.load_from_file = load_from_file
+        self.meta_file = meta_file
+        if self.data_root is not None:
+            if not osp.isabs(self.meta_file):
+                self.meta_file = osp.join(self.data_root, self.meta_file)
+        self.filter_labels = filter_labels
+        self.rank, self.world_size = get_dist_info()
+        self.temp_img_metas = []
+        self.test_img_metas = []
+        self.test_img_shapes = []
+        self.load_from_pipeline = False if load_from_file else True
+
+    def get_classes_from_csv(self, label_description_file):
+        """Get class name and label map proto.
+
+        Args:
+            label_description_file (str): File path to the label map proto.
+
+        Returns:
+            list[str]: Class name of OpenImages.
+        """
+
+        index_list = []
+        classes_names = []
+        with open(label_description_file, 'r') as f:
+            reader = csv.reader(f)
+            for line in reader:
+                self.cat2label[line[0]] = line[1]
+                classes_names.append(line[1])
+                index_list.append(line[0])
+        self.index_dict = {index: i for i, index in enumerate(index_list)}
+        return classes_names
+
+    def list_from_csv(self, ann_file):
+        """Load annotation from csv style ann_file.
```

Comment:
ann_file -> annotation file. Should make the function summary a complete sentence.